targets/csharp/source/source/Extensions/Models/PlayFabCloudScriptModelsExtensions.cs.ejs

```diff
@@ -17,55 +24,222 @@ namespace PlayFab.CloudScriptModels
 
         public TFunctionArgument FunctionArgument { get; set; }
 
-        protected FunctionExecutionContext() { }
+        public string CurrentPlayerId { get; set; }
+
+        protected FunctionContext() { }
 
         /// <summary>
-        /// Creates a new <c>FunctionExecutionContext</c> out of the incoming request to an Azure Function.
+        /// Creates a new <c>FunctionContext</c> out of the incoming request to an Azure Function.
         /// </summary>
         /// <param name="request">The request incoming to an Azure Function from the PlayFab server</param>
-        /// <returns>A new populated <c>FunctionExecutionContext</c> instance</returns>
-        public static async Task<FunctionExecutionContext<TFunctionArgument>> Create(System.Net.Http.HttpRequestMessage request)
+        /// <returns>A new populated <c>FunctionContext</c> instance</returns>
+        public static async Task<FunctionContext<TFunctionArgument>> Create(HttpRequestMessage request)
         {
             using (var content = request.Content) 
             {
                 var body = await content.ReadAsStringAsync();
-                var contextInternal = Json.PlayFabSimpleJson.DeserializeObject<FunctionExecutionContextInternal>(body);
+                var contextInternal = Json.PlayFabSimpleJson.DeserializeObject<FunctionContextInternal>(body);
```

Comment:
You are hard coding yourself to a fixed json serializer, and that's not what you should be doing.

                var json = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer);
                var contextInternal = json.DeserializeObject<FunctionContextInternal>(body);